src/apps/basic/basic_apps.lua

```diff
@@ -86,6 +87,90 @@ function Sink:push ()
    end
 end
 
+--- ### `Match` app: Compare packets recieved on rx and comparator
+Match = {}
+
+function Match:new(cfg)
+   if cfg == "nil" then
+      cfg = { mode = "exact" }
+   end
+   assert(cfg.mode == "exact" or cfg.mode == "monotonic",
+      ("Unsupported mode %s"):format(cfg.mode))
+   return setmetatable({
+         cfg = cfg,
+         seen = 0,
+         errs = { },
+         done = false
+      },
+      {
+         __index=Match
+      })
+end
```

Comment:
`cfg` is being converted to a string here because Apps are supposed to be able to accept their configuration as strings. The common pattern to read an app config is this:

```lua
function App:new (arg)
      local conf = config.parse_app_arg(arg) or {} -- The default
      ....
end
```